controllers/solrprometheusexporter_controller.go

```diff
@@ -209,7 +209,22 @@ func (r *SolrPrometheusExporterReconciler) Reconcile(req ctrl.Request) (ctrl.Res
 		}
 	}
 
-	deploy := util.GenerateSolrPrometheusExporterDeployment(prometheusExporter, solrConnectionInfo, configXmlMd5, tlsClientOptions)
+	basicAuthMd5 := ""
+	if prometheusExporter.Spec.SolrReference.BasicAuthSecret != nil {
+		selector := prometheusExporter.Spec.SolrReference.BasicAuthSecret
+		basicAuthSecret := &corev1.Secret{}
+		err := r.Get(context.TODO(), types.NamespacedName{Name: selector.Name, Namespace: prometheusExporter.Namespace}, basicAuthSecret)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		passBytes, ok := basicAuthSecret.Data[selector.Key]
```

Comment:
We might have to look in two places here. It could be in `data` or `stringData`.

https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret

Reply:
afaik, `stringData` is a convenience for YAML and secrets always store all data in `data` but I'll verify if they supply via `stringData` that we can read it from `data`

Reply:
Ahhh very cool, did not know that!

Reply:
Confirmed that only `data` is stored ... here's an example:
```
apiVersion: v1
stringData:
  k8s-oper: Test1234
kind: Secret
metadata:
  name: user-provided-basic-auth
type: Opaque
```
and `kubectl get secret user-provided-basic-auth -o yaml`  gives:
```
apiVersion: v1
data:
  k8s-oper: VGVzdDEyMzQ=
kind: Secret
metadata:
  name: user-provided-basic-auth
type: Opaque
```
and in the logs, we have:
```
2021-02-19T19:38:56.486Z	INFO	controllers.SolrCloud	For user-supplied secret, password is:	{"namespace": "dev", "solrCloud": "dev", "password": "Test1234"}
```
_Note: I just added that log message to test this scenario ... it has since been removed, i.e. we're not logging passwords_